### PR TITLE
WS2-1526 - Pathauto + User account URLs

### PR DESCRIPTION
**Pull request by bjdevil21**
This adds a default Pathauto pattern for all users' URLs (/user/ASURITEID) instead of the default /user/12345. It also applies the URL change when database updates are applied.

### CODE REVIEW

**Comment by bjdevil21** [comment]
Will be reviewed at 330PM today. Updated .gitignore to match config diffs bash script changes.

**Review by mlsamuelson — CHANGES_REQUESTED** [review]
@bjdevil21 In Integrated Search we already have user profile paths we use and we don't want to lose that as the destination the user path resolves to in the browser. I suspect others have similar needs.  

Let's instead keep the config so it gets established on NEW site install, but then instead of automatically running the settings update and batch API cleanup for existing sites - let's make that optional by adding a page in the admin UI (maybe the "developer" settings under "configuration") with a button to kick off the batch API - it can update the pathauto then, and then take advantage of the existing batch API code.

**Comment by bjdevil21** [comment]
I saw your Slack comments. I have removed the part of the DB update that adds aliases to existing URLs.

A quick release note write up with a, "Go to /admin/config/search/path/update_bulk" and follow some short, explicit instructions and they won't need anything else (like a one-off "apply aliases to your users" button).

Cameo already verified that the install step works after a bug fix this afternoon (I was trying to update an existing config vs. import a new one). We'll recheck it tomorrow, though, in this state.

**Comment by bjdevil21** [comment]
@mlsamuelson - Update: I went through and put in some safeguards to keep URLs safe or causing any trouble:

- **As a new install**: The pattern is installed and will be applied to all users (except UID 1) as they are created.
- **As an update to existing sites**:
-- If there's an existing pattern for users, then the new pattern is still installed but comes last in priority (by Drupal weight) and will be ignored.
-- If there's not an existing pattern for users, then the Batch API processes the URLs to add aliases - but it will only be applied **IF** it doesn't already have an alias. (Doesn't matter how the alias was set - manually entered or applied by a PA pattern. And if the "auto apply alias" checkbox is unchecked, then the alias will not apply - even if the alias field is empty.)

**In short, the URL pattern is applied but comes last, and aliases are generated only if the user don't have an alias.**

If you want, we can run a test of the code on a multidev of a site with a huge users dataset?

(FYI - There was a super easy way to have this update apply an alternative alias alongside the existing ones, but for sites with thousands of users we don't want to force the bloating of the path_% DB tables.)

File: config/install/pathauto.pattern.user_profile_urls.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+dependencies:
+  module:
+    - user
+id: user_profile_urls
+label: 'User Profile URLs'
+type: 'canonical_entities:user'
+pattern: 'user/[user:account-name]'
+selection_criteria: {  }
+selection_logic: and
+weight: 0
```
Comment (mlsamuelson) [inline config/install/pathauto.pattern.user_profile_urls.yml:12]: We, and possibly other projects, have implemented our own user path settings, and the webspark.install code that loads in this new pathauto pattern creates a competing rule that we want to ensure doesn't take precedence over existing rules.  

As per the behaviors you see depicted regarding pathauto weight in configs in [this Drupal.org issue](https://www.drupal.org/project/pathauto/issues/2839643), can you set the weight here to 5 or something, so if an existing  rule with weight 0 exists, we don't have unexpected tie-breaker behavior?

**Review by mlsamuelson — CHANGES_REQUESTED** [review]

**Comment by mlsamuelson** [comment]
Conflict resolution needed here, too. Will be impacted by PR #72 as well.  Perhaps let's get that one in first, and then this can be resolved.

**Comment by bjdevil21** [comment]
I had already resolved the weight issue in a different branch (it was set to weight: 10 in the ws2-1526-batch), but I hadn't merged it into this branch. That is done now. 

The .install file conflict from PR #73 has also been resolved.

**Review by mlsamuelson — APPROVED** [review]
